lib/ecmascript.ts

```diff
@@ -1294,798 +1484,489 @@ export const ES = ObjectAssign({}, ES2020, {
+  if (result !== undefined) {
+    result = ToString(result);
+  }
+  return result;
+}
+
+export function CalendarEraYear(calendar, dateLike) {
+  let result = calendar.eraYear(dateLike);
+  if (result !== undefined) {
+    result = ToIntegerThrowOnInfinity(result);
+  }
+  return result;
+}
+
+export function CalendarDayOfWeek(calendar, dateLike) {
+  return calendar.dayOfWeek(dateLike);
+}
+
+export function CalendarDayOfYear(calendar, dateLike) {
+  return calendar.dayOfYear(dateLike);
+}
+
+export function CalendarWeekOfYear(calendar, dateLike) {
+  return calendar.weekOfYear(dateLike);
+}
+
+export function CalendarDaysInWeek(calendar, dateLike) {
+  return calendar.daysInWeek(dateLike);
+}
+
+export function CalendarDaysInMonth(calendar, dateLike) {
+  return calendar.daysInMonth(dateLike);
+}
+
+export function CalendarDaysInYear(calendar, dateLike) {
+  return calendar.daysInYear(dateLike);
+}
+
+export function CalendarMonthsInYear(calendar, dateLike) {
+  return calendar.monthsInYear(dateLike);
+}
+
+export function CalendarInLeapYear(calendar, dateLike) {
+  return calendar.inLeapYear(dateLike);
+}
+
+export function ToTemporalCalendar(calendarLike) {
+  if (Type(calendarLike) === 'Object') {
+    if (HasSlot(calendarLike, CALENDAR)) return GetSlot(calendarLike, CALENDAR);
+    if (!('calendar' in calendarLike)) return calendarLike;
+    calendarLike = calendarLike.calendar;
+    if (Type(calendarLike) === 'Object' && !('calendar' in calendarLike)) return calendarLike;
+  }
+  const identifier = ToString(calendarLike);
+  const TemporalCalendar = GetIntrinsic('%Temporal.Calendar%');
+  if (IsBuiltinCalendar(identifier)) return new TemporalCalendar(identifier);
+  let calendar;
+  try {
+    ({ calendar } = ParseISODateTime(identifier, { zoneRequired: false }));
+  } catch {
+    throw new RangeError(`Invalid calendar: ${identifier}`);
+  }
+  if (!calendar) calendar = 'iso8601';
+  return new TemporalCalendar(calendar);
+}
+
+export function GetTemporalCalendarWithISODefault(item) {
+  if (HasSlot(item, CALENDAR)) return GetSlot(item, CALENDAR);
+  const { calendar } = item;
+  if (calendar === undefined) return GetISO8601Calendar();
+  return ToTemporalCalendar(calendar);
+}
+
+export function CalendarCompare(one, two) {
+  const cal1 = ToString(one);
+  const cal2 = ToString(two);
+  return cal1 < cal2 ? -1 : cal1 > cal2 ? 1 : 0;
+}
+
+export function CalendarEquals(one, two) {
+  if (one === two) return true;
+  const cal1 = ToString(one);
+  const cal2 = ToString(two);
+  return cal1 === cal2;
+}
+
+export function ConsolidateCalendars(one, two) {
+  if (one === two) return two;
+  const sOne = ToString(one);
+  const sTwo = ToString(two);
+  if (sOne === sTwo || sOne === 'iso8601') {
+    return two;
+  } else if (sTwo === 'iso8601') {
+    return one;
+  } else {
+    throw new RangeError('irreconcilable calendars');
+  }
+}
+
+export function DateFromFields(calendar, fields, options?: any) {
```

Comment:
Should the options type be `Temporal.AssignmentOptions` ?  That's what's used in index.d.ts, although if that's not accurate then we should fix it.